feat: SRC-14 owned proxy usage scripts and docs (#8)
FuelLabs/sway-standard-implementations: src14/owned_proxy/README.md

# SRC-14: Owned Proxy Contract

- The [Owned Proxy Contract](./contract/src/main.sw) is an opinionated implementation of the [extended SRC-14 standard](https://docs.fuel.network/docs/sway-standards/src-14-simple-upgradeable-proxies/). It utilises the [Upgradability library from sway-libs](https://github.com/FuelLabs/sway-libs) and includes initialization functionality that allows for secure ownership upon deployment.

## Usage instructions

Clone the repository and move into it:

```bash
git clone -b master https://github.com/FuelLabs/sway-standard-implementations.git && cd sway-standard-implementations
```

### Rust scripts

A suite of Rust scripts are included to assist in the usage of the contract. In order to use them navigate to the [owned-proxy/scripts](./scripts/) directory. From `sway-standard-implementations/<you are here>`:

```bash
cd src14/owned_proxy/scripts
```

#### Deploy and initialize

> To deploy and initialize the proxy contract the `deploy_and_init` script is available. It will use the arguments `--initial-target` and `--initial-owner` as configurables in the proxy contract. From `sway-standard-implementations/src14/owned_proxy/scripts/<you are here>`:

Reviewer: - We need to add a section to inform you that the user needs to perform a deployment of the contract first, which will be used as a target.
- We can also inform you that it is possible to deploy the contract after and then set the target.
- Change the default URL to localhost as a good practice.

# SRC-14: Owned Proxy Contract

- The [Owned Proxy Contract](./contract/src/main.sw) is an opinionated implementation of the [extended SRC-14 standard](https://docs.fuel.network/docs/sway-standards/src-14-simple-upgradeable-proxies/). It utilises the [Upgradability library from sway-libs](https://github.com/FuelLabs/sway-libs) and includes initialization functionality that allows for secure ownership upon deployment.

> 

Reviewer: Lets add a warning here saying that this utils scripts are temporary and will be integrated into forc in the near future.

# SRC-14: Owned Proxy Contract

- The [Owned Proxy Contract](./contract/src/main.sw) is an opinionated implementation of the [extended SRC-14 standard](https://docs.fuel.network/docs/sway-standards/src-14-simple-upgradeable-proxies/). It utilises the [Upgradability library from sway-libs](https://github.com/FuelLabs/sway-libs) and includes initialization functionality that allows for secure ownership upon deployment.

## Usage instructions

Clone the repository and move into it:

```bash
git clone -b master https://github.com/FuelLabs/sway-standard-implementations.git && cd sway-standard-implementations
```

### Rust scripts

> > **Warning:** These scripts provide a temporary utility. Proxy contract deployment and interaction will be integrated into `Forc` in the near future.

Reviewer: ```suggestion
> **Warning:** These scripts are temporary. Proxy contract deployment and interactions will be integrated into `Forc` in the near future.
```
